opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1263,8 +1281,16 @@ namespace Opm {
             std::vector<int> failed_cells_pb;
             std::vector<int> failed_cells_pd;
 
+            if ( ebosModel.cachedIntensiveQuantities(/*cellIdx=*/0, /*timeIdx=*/0) == nullptr ) {
+                updateSimCache();
+            }
+
             for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
```

Comment:
instead of looping over the cell indices here and retrieving the intensive quantities from the cache, it is better to loop over the elements and let the element context handle the intensive quantities. that way it also works in the hypothetical case the intensive quantities do not get cached...